### PR TITLE
Component refactor

### LINKED ISSUE #234: Component consistency
There's no consistency among components - some are initializated as `playerList.init(bot)`, some are like `hotbar.bot = bot`

Maybe we should also put them into `components/` directory?

Imo, this needs refactoring (maybe including CSS instead of putting them into JS)

### PR DESCRIPTION
**Pull request by Saiv46**
Closes #234

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-require('./lib/healthbar')
-require('./lib/hotbar')
-require('./lib/gameMenu')
-require('./lib/chat')
-require('./lib/crosshair')
-require('./lib/playerlist')
-require('./lib/debugmenu')
+require('./src/menu')
+require('./src/loading_screen')
+require('./src/hotbar')
```
Comment (rom1504) [inline index.js:4]: why is this not a component?

File: src/lib/loadingscreen/index.css [hunk 2]
```diff
@@ -0,0 +1,32 @@
+h1 {
+	font-family: mojangles, minecraft, monospace;
+}
+.loader {
+	display: initial;
```
Comment (rom1504) [inline src/lib/loadingscreen/index.css:5]: As discussed, I'd prefer not separating css and html and js in different files
Is there a benefit?

**Comment by rom1504** [comment]
General comments:
* Prefer doing several PRs for several unrelated changes to ease reviews
* avoid moving files around if there is no real objective
* why split html/js/css in different files ? That's not the spirit of components
* prefer fixing the actual issues with the code rather than moving things around (like reducing the amount of code and making components more components like) if you want to work on refactoring

**Comment by Saiv46** [comment]
> General comments:
> 
> * Prefer doing several PRs for several unrelated changes to ease reviews
There's no unrelated changes (expect for chat, I'll undo that when PR gonna leave draft stage).

> * why split html/js/css in different files ? That's not the spirit of components
It would ease things when it'll be decided to utilize Webpack features like lazy-loading, CSS preprocessing with loaders and more. And also I hate when CSS-in-JS are made for no reason. For example, HTML/JS/CSS in Vue Singe File Components are still splitted during build.

> * prefer fixing the actual issues with the code rather than moving things around (like reducing the amount of code and making components more components like) if you want to work on refactoring
Isn't that I'm preparing to do so?

**Comment by AwesomestCode** [comment]
Quick question, what was the reason for closing?

**Comment by Saiv46** [comment]
> Quick question, what was the reason for closing?

Disagreement about whatever CSS should be inside or outside of components.

Also, this PR is huge and hard to review

**Pull request closed without merging**